### PR TITLE
README.md: update to reflect changes in operator-sdk

### PR DESCRIPTION
**Pull request by estroz**
This PR brings the `getting-started` user guide in sync with `operator-sdk`'s user guide.

@ecordell can you make sure the OLM-related portion is up to date (perhaps in a separate PR)? Thanks!

Resolves: #13, #15, #16, #18, #26

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,158 +1,339 @@
...
+Add a new [Controller][controller_go_doc] to the project that will watch and reconcile the `Memcached` resource:
+
+```sh
+$ operator-sdk add controller --api-version=cache.example.com/v1alpha1 --kind=Memcached
 ```
 
-#### Define the Handler
+This will scaffold a new Controller implementation under `pkg/controller/memcached/...`.
+
+For this example replace the generated Controller file `pkg/controller/memcached/memcached_controller.go` with the example [`memcached_controller.go`][memcached_controller] implementation.
```
Comment (hasbro17) [inline README.md]: The `memcached_controller` link doesn't exist. This should be pointing to the file that replaces the handler.

File: README.md [hunk 2]
```diff
@@ -1,158 +1,339 @@
...
+
+### 1. Run as a Deployment inside the cluster
+
+Build the memcached-operator image and push it to a registry:
 
 ```sh
-$ cd $GOPATH/src/github.com/example-inc/
-$ cd memcached-operator
 $ operator-sdk build quay.io/example/memcached-operator:v0.0.1
+$ sed -i 's|REPLACE_IMAGE|quay.io/example/memcached-operator:v0.0.1|g' deploy/operator.yaml
```
Comment (AlexNPavel) [inline README.md]: Maybe add the note about OSX? https://github.com/operator-framework/operator-sdk/pull/699

File: README.md [hunk 3]
```diff
@@ -200,18 +383,26 @@ memcached-for-wordpress-65b75fd8c9-7b9x7   1/1       Running   0          14m
...
+[doc_sdk_install_instr]: https://github.com/operator-framework/operator-sdk/blob/master/doc/user-guide.md#install-the-operator-sdk-cli
+[doc_olm_install_instr]: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/install/install.md
 [layout_doc]: https://github.com/operator-framework/operator-sdk/blob/master/doc/project_layout.md
-[quay_io]: https://www.quay.io
-[kubernetes_cr]: https://kubernetes.io/docs/concepts/api-extension/custom-resources/
-[handler_go]: https://github.com/operator-framework/getting-started/blob/master/handler.go.tmpl#L7
-[create_public_image]: https://quay.io/new/
+[manager_go_doc]: https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/manager#Manager
+[controller_go_doc]: https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg#hdr-Controller
+[memcached_controller]: https://github.com/operator-framework/operator-sdk-samples/blob/master/memcached-operator/pkg/controller/memcached/memcached_controller.go
```
Comment (hasbro17) [inline README.md]: We should point to the example controller used specifically for the user-guide.
https://github.com/operator-framework/operator-sdk/blob/master/example/memcached-operator/memcached_controller.go.tmpl

The samples repo memcached-operator has a few differences like the import path being different from what we use in the example.

**Pull request merged**